app/uk/gov/hmrc/trusts/controllers/SubmissionDraftController.scala

```diff
@@ -0,0 +1,212 @@
+                    draftData = newDraftData,
+                    reference = newReference,
+                    inProgress = newInProgress)
+                  submissionRepository.setDraft(newDraft).map(
+                    result => if (result) {
+                      Ok
+                    } else {
+                      InternalServerError
+                    }
+                  )
+                case e: JsError => Future.successful(InternalServerError(e.errors.toString()))
+              }
+
+            }
+          )
+        case _ => Future.successful(BadRequest)
+      }
+    }
+  }
+
+  // Play 2.5 throws if the path to be pruned does not exist.
+  // So we do this hacky thing to keep it all self-contained.
+  // If upgraded to play 2.6, this can turn into simply "path.json.prune".
+  private def prunePath(path: JsPath) = {
+    JsPath.json.update {
+      path.json.put(Json.toJson(Json.obj()))
+    } andThen path.json.prune
+  }
+
+  private def setRegistrationSection(path: String, registrationSectionData: JsValue) : Reads[JsObject] = {
+    val sectionPath = (JsPath \ "registration" \ path)
+    prunePath(sectionPath) andThen JsPath.json.update(sectionPath.json.put(registrationSectionData))
+  }
+
+  private def setRegistrationSections(pieces: List[RegistrationSubmissionDraftPiece]) : List[Reads[JsObject]] = {
+    pieces.map(piece => setRegistrationSection(piece.elementPath, piece.data))
+  }
+
+  private def setStatus(statusOpt: Option[RegistrationSubmissionDraftStatus]): Reads[JsObject] = {
+    statusOpt match {
+      case Some(draftStatus) =>
+        val sectionPath = (JsPath \ "status" \ draftStatus.section)
+
+        draftStatus.status match {
+          case Some(status) => prunePath(sectionPath) andThen JsPath.json.update(sectionPath.json.put(Json.toJson(status.toString)))
+          case None => prunePath(sectionPath)
+        }
+      case _ => JsPath.json.pick[JsObject]
+    }
+  }
+
+  def setSectionSet(draftId: String, sectionKey: String): Action[JsValue] = identify.async(parse.json) {
+    implicit request => {
+      request.body.validate[RegistrationSubmissionDraftSetData] match {
+        case JsSuccess(incomingDraftData, _) =>
+          submissionRepository.getDraft(draftId, request.identifier).flatMap(
+            result => {
+              val draft: RegistrationSubmissionDraft = result match {
+                case Some(draft) => draft
+                case None => RegistrationSubmissionDraft(draftId, request.identifier, localDateTimeService.now, Json.obj(), None, Some(true))
+              }
+
+              val sectionPath = JsPath() \ sectionKey
+
+              val thingsToDo = List(
+                prunePath(sectionPath),
+                JsPath.json.update {
+                  sectionPath.json.put(Json.toJson(incomingDraftData.data))
+                },
+                setStatus(incomingDraftData.status)
+              ) ++ setRegistrationSections(incomingDraftData.registrationPieces)
```

Comment:
What does this do? :D

Reply:
It is updating three things: the Json for the section (typically user answers, private to individual front end), the status for the section (task list in trusts-frontend), and then all the individual registration section pieces, which come in as an array (submitted data in trusts-frontend). Eventually, there will be the print page rows as well.